### DISCUSSION TITLE
Preserve Slashes in params Without Encoding to %2F

### DISCUSSION THREAD

**Discussion by MrPaulAR** [encode/httpx#1668] [Q&A]
When I execute the snippet below the slashes in the params are being converted to %2F which is breaking the API calls.

```
url = f"/dashboard/appliance/{device_name}/live"
params = {"command": f"bgp/neighbors/{routing-instance}"}
r = self.client.get(url, params=params)
r.raise_for_status()
print r.json()
```
What is being sent is the following which results in a 404.

> https://host/dashboard/appliance/device/live?command=bgp%2Fneighbors%2FMgmt

Is there a way to prevent this from being converted?

**Comment by lovelydinosaur** [comment]
Nope, we use the same behaviour as [Python's `urlencode`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlencode).

I don't think you'd be able to get unquoted slashes with `requests` either?

You *can* preserve the style if you construct the URL string explicitly, tho, for example....

```python
url = f"/dashboard/appliance/{device_name}/live?command=bgp/neighbors/{routing-instance}"
```
